crates/autopilot/src/solvable_orders.rs

```diff
@@ -168,17 +169,25 @@ impl SolvableOrdersCache {
         let mut counter = OrderFilterCounter::new(self.metrics, &db_solvable_orders.orders);
 
         let orders = filter_banned_user_orders(db_solvable_orders.orders, &self.banned_users);
-        counter.checkpoint("banned_user", &orders);
+        let removed = counter.checkpoint("banned_user", &orders);
+        self.database
+            .store_order_events(removed, OrderEventLabel::Invalid);
```

Comment:
Seeing this repetition, does it makes sense to add a:

```rust
counter.checkpoint_with_events("banned_user", &orders, OrderEventLabel::Invalid);
```

To prevent accidentally adding new checkpoints and forgetting to store the order events in the DB?

Reply:
I also though about that but then you either have to make the DB a member of `OrderFilterCounter` or have a helper function on `SolvableOrdersCache` which takes a lot of arguments. Both seem kind of weird and we already have the problem of being able to forget creating a checkpoint after introducing a new filter so I'm not sure this is worth it.